assets/Product.wxs

```diff
@@ -165,6 +166,10 @@
             <Component Id="Telemetry" Guid="{80520f20-471d-4d08-adc8-bab637627b39}" KeyPath="yes">
               <Environment Id="PowerShellDistributionChannel" Action="create" Name="POWERSHELL_DISTRIBUTION_CHANNEL" Permanent="no" System="yes" Value="MSI:[WINDOWS_PRODUCT_NAME]"/>
             </Component>
+            <Component Id="DisableTelemetry" Guid="{E86F5C91-3721-4C5C-A31F-1F0A5A468573}" KeyPath="yes">
+              <Condition>DISABLE_TELEMETRY</Condition>
+              <Environment Id="PowerShellTelemetryOptout" Action="set" Name="POWERSHELL_TELEMETRY_OPTOUT" Permanent="no" System="yes" Value="1"/>
```

Comment:
Do you want this to default to the env value if they already have it set? Schedule a type 51 custom action early in both InstallUISequence and InstallExecuteSequence tables - probably right after AppSearch:

```xml
<SetProperty Id="DISABLE_TELEMETRY" Value="[%POWERSHELL_TELEMETRY_OPTOUT]" After="AppSeach" Sequence="first" />
```

Reply:
I don't think it is applicable in this scenario because `DISABLE_TELEMETRY` will and should be set just by the user (either via the UI checkbox or as a parameter passed to msiexec). The logic is: If the user wants to opt out of telemetry, this env variable shall be set to 1 (that's what needs to happen to disable telemetry for PowerShell), if the user does not want to opt out, then no change to the env variable is made, no matter the current state because the user might have set that env variable already beforehand deliberately in order to opt out of telemetry and due to defaults, this would be a breaking change to them and could accidentally enable telemetry for them, hence why I don't want to do any action if the user has not chosen (either deliverately or due to defaults) to not opt out of telemetry.

Reply:
That checkbox will default to the un/checked state each time the installer is started. What I'm suggesting is to set the property that dictates the state to the current environment variable, if any. If unset, the default check state should still be the same (you can test this - I haven't done MSI UI in over a decade because chainers are more flexible and necessary for multi-package scenarios, which is why we wrote WiX Burn).

If by being unset the default check state isn't right, you could also schedule another type 51 custom action (which is what `SetProperty` creates and schedules) to set it to the right value if the previous value is unset.

Or maybe it's not worth it. This was only a suggestion to retain the customers' current value, since MSI properties are not retained by default. There are methods to persist them using registry, but in this case you already have a persistent store: the current environment, so no need to have it in two places and chance them getting out of sync.

Reply:
Oh, I understand now what you mean. I think pre-populating the checkbox could have a negative side-effect: If the user has already set the environment variable and the MSI installer then ticks it by default and the user then uninstalls powershell (maybe because it was a preview), then the uninstall would remove the environment variable and would be undesired. Therefore I think it is better to be explicit and only perform actions when the user explicitly asks for it.